controllers/etcddefrag_controller.go

```
@@ -0,0 +1,682 @@
+	c, backends, err := r.dialAndProbe(ctx, cluster, running)
+	if err != nil {
+		// Without a client we can neither defragment nor disarm; if the run has
+		// outlived its deadline while stuck here, fail it rather than requeue
+		// forever.
+		if exceededDeadline(df) {
+			return r.fail(ctx, df, "DeadlineExceeded", deadlineMsg(df))
+		}
+		logger.Error(err, "defrag: cannot dial/probe cluster; retrying")
+		return ctrl.Result{RequeueAfter: defragRequeueAfter}, nil
+	}
+	defer c.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `c.Close` is not checked

(errcheck)